framework/db/mysql/Schema.php

```diff
@@ -596,4 +602,105 @@ private function loadTableConstraints($tableName, $returnType)
 
         return $result[$returnType];
     }
+
+    /**
+     * Detect if a column has a default value in form of expression
+     * @param $extra string 'Extra' detail obtained from "SHOW FULL COLUMNS ...". Example: 'DEFAULT_GENERATED'
+     * @return bool true if the column has default value in form of expression instead of constant
+     * @see https://github.com/yiisoft/yii2/issues/19747
+     * @see https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html
+     * @since 2.0.48
+     */
+    public function defaultIsExpression($info)
+    {
+        if ($this->isMysql()) {
+            // https://dev.mysql.com/doc/refman/5.7/en/information-schema-columns-table.html and
+            // https://dev.mysql.com/doc/refman/8.0/en/information-schema-columns-table.html
+            return (
+                // for MySQL >= 8
+                (strpos($info['extra'], static::DEFAULT_EXPRESSION_IDENTIFIER) !== false) ||
+                (strpos($info['extra'], static::CURRENT_TIMESTAMP_DEFAULT_EXPRESSION_IDENTIFIER) !== false) ||
+                // for MySQL < 8
+                (strpos($info['default'], static::CURRENT_TIMESTAMP_DEFAULT_EXPRESSION_IDENTIFIER) !== false &&
+                    ((strpos($info['type'], 'datetime') !== false) || (strpos($info['type'], 'timestamp') !== false))
+                )
+            );
+        } else { // MariaDB
+            // There is no strong way in MariaDB to detect default value is constant or expression. This is implemented on the basis pattern observed from data in Information_schema.Extra
+            $moreInfo = $this->moreColumnInfo($info['field']);
```

Comment:
provide table name as an argument to this function instead of relying on the value of a private property.